File: frontend/src/app/api-generated/models/booking-place-view-model.ts
/* tslint:disable */
/* eslint-disable */
import { BookingPlaceItemTypeEnum } from '../models/booking-place-item-type-enum';
import { BookingPlaceTypeEnum } from '../models/booking-place-type-enum';
import { BookingStateEnum } from '../models/booking-state-enum';
import { UserViewModel } from '../models/user-view-model';
export interface BookingPlaceViewModel {
  availableForBooking?: boolean;
  availableFrom?: string | null;
  availableTo?: string | null;
  floorId?: number;
  id?: number;
  itemType?: BookingPlaceItemTypeEnum;
  name?: string | null;
  reservedForUserId?: number | null;
  reservedForUserVm?: UserViewModel;
  state?: BookingStateEnum;
  type?: BookingPlaceTypeEnum;
}

File: backend/ViewModels/BookingPlaceViewModel.cs
﻿using BookingApp.Enums;

namespace BookingApp.ViewModels
{
    public class BookingPlaceViewModel
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public BookingPlaceTypeEnum Type { get; set; }
        public BookingPlaceItemTypeEnum ItemType { get; set; }
        public BookingStateEnum State { get; set; }
        public bool AvailableForBooking { get; set; }
        public DateTime? AvailableFrom { get; set; }
        public DateTime? AvailableTo { get; set; }
        public int FloorId { get; set; }
        public int? ReservedForUserId { get; set; }
        public UserViewModel? ReservedForUserVm { get; set; }
    }
}
